pennylane/capture/primitives.py

```diff
@@ -117,3 +202,122 @@ def _(*_, **__):
         return abstract_type()
 
     return primitive
+
+
+def create_measurement_obs_primitive(
+    measurement_type: Type["qml.measurements.MeasurementProcess"], name: str
+) -> Optional["jax.core.Primitive"]:
+    """Create a primitive corresponding to the input type where the abstract inputs are an operator.
+
+    Called by default when defining any class inheriting from :class:`~.MeasurementProcess`, and is used to
+    set the ``MeasurementProcesss._obs_primitive`` property.
+
+    Args:
+        measurement_type (type): a subclass of :class:`~.MeasurementProcess`
+        name (str): the preferred string name for the class. For example, ``"expval"``.
+            ``"_obs"`` is appended to this name for the name of the primitive.
+
+    Returns:
+        Optional[jax.core.Primitive]: A new jax primitive. ``None`` is returned if jax is not available.
+
+    """
+    if not has_jax:
+        return None
+
+    primitive = jax.core.Primitive(name + "_obs")
+
+    @primitive.def_impl
+    def _(obs, **kwargs):
+        return type.__call__(measurement_type, obs=obs, **kwargs)
+
+    abstract_type = _get_abstract_measurement()
+
+    @primitive.def_abstract_eval
+    def _(*_, **__):
+        abstract_eval = measurement_type._abstract_eval  # pylint: disable=protected-access
+        return abstract_type(abstract_eval, n_wires=None)
```

Comment:
I'm guessing the abstract evaluation will actually be invoked with the "capture.measure" primitive?

Reply:
Never mind I'm seeing this is now handled by the qnode primitive :)